Sources/RibbonKit/List/RibbonListViewDiffableDataSource.swift

```diff
@@ -16,9 +16,19 @@ open class RibbonListViewDiffableDataSource<Section: Hashable, Item: Hashable>:
         self._ribbonList = ribbonList
         super.init()
 
+        let sectionLeadingCellRegistration = UICollectionView.CellRegistration<RibbonListSectionLeadingCell, Item> { _, _, _ in }
+
         dataSource = UICollectionViewDiffableDataSource<Section, Item>(collectionView: _ribbonList.collectionView, cellProvider: {
             [unowned self] collectionView, indexPath, itemIdentifier in
-            cellProvider(_ribbonList, indexPath, itemIdentifier)
+            guard indexPath.item == 0,
+                  let leadingCellView = _ribbonList.viewForLeadingCell(inSection: indexPath.section) else {
+                return cellProvider(_ribbonList, indexPath, itemIdentifier)
+            }
+            let cell = collectionView.dequeueConfiguredReusableCell(using: sectionLeadingCellRegistration, for: indexPath, item: itemIdentifier)
+            cell.setView(leadingCellView)
+            cell.hideContentView = true
+            _ribbonList.sectionsWithLeadingCellComponent.insert(indexPath.section)
```

Comment:
In the scenario that once a view was returned, the section is added to this set, and then no view is returned during lifecycle of the ribbon list, shouldn't the section be removed from the set?

Reply:
I just pushed a new commit that aims to solve what I think you meant. Let me know what you think